dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorPower.java

```diff
@@ -438,10 +437,9 @@ public List<Event> visitRMWOpReturn(RMWOpReturn e) {
 		String mo = e.getMo();
 		
         Register dummyReg = e.getThread().newRegister(resultRegister.getPrecision());
-        Local localOp = newLocal(resultRegister, new IExprBin(dummyReg, op, value));
-
         // Power does not have mo tags, thus we use null
         Load load = newRMWLoadExclusive(dummyReg, address, null);
+        Local localOp = newLocal(resultRegister, new IExprBin(dummyReg, op, value));
         Store store = newRMWStoreExclusive(address, resultRegister, null, true);
```

Comment:
Here you might have an issue if `address` depends on `resultRegister`. You should probably update the dummy with the localOp, then perform the store and then set `resultRegister=dummy`.